### PR TITLE
RPG titles wizard event

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Based off an event melbert ran, where each person had their name and a level:

image

So I adapted it into a full feature:

image

image

## Why It's Good For The Game

When melbert ran this event everyone had some good jolly fun, we can now officially go even deeper into the RPG rabbit hole, yeeeeeeap

## Changelog

:cl: tralezab code, melbert idea!
add: There's a new wizard event, RPG titles. Summon the star duke!
/:cl:

### CODE REVIEW

**Comment by AMonkeyThatCodes** [comment]
is the text more consistently placed than in the events?

**Comment by tralezab** [comment]
It should be? But you know how maptext is.

File: code/modules/jobs/job_types/prisoner.dm [hunk 1]
```diff
@@ -20,7 +20,7 @@
 	)
 
 	family_heirlooms = list(/obj/item/pen/blue)
-
+	rpg_title = "Defeated Miniboss"
```
Comment (AMonkeyThatCodes) [inline code/modules/jobs/job_types/prisoner.dm:23]: Future Protagonist might be better

File: code/modules/jobs/job_types/head_of_security.dm [hunk 2]
```diff
@@ -32,7 +32,7 @@
 	bounty_types = CIV_JOB_SEC
 
 	family_heirlooms = list(/obj/item/book/manual/wiki/security_space_law)
-
+	rpg_title = "Guard Leader"
```
Comment (AMonkeyThatCodes) [inline code/modules/jobs/job_types/head_of_security.dm:35]: Head of the Guard

File: code/modules/jobs/job_types/head_of_personnel.dm [hunk 3]
```diff
@@ -36,7 +36,7 @@
 	)
 
 	family_heirlooms = list(/obj/item/reagent_containers/food/drinks/trophy/silver_cup)
-
+	rpg_title = "Guild Questgiver"
```
Comment (AMonkeyThatCodes) [inline code/modules/jobs/job_types/head_of_personnel.dm:39]: Guildmaster?

File: code/modules/jobs/job_types/head_of_security.dm (hunk 2 above)
Comment (tralezab) [reply code/modules/jobs/job_types/head_of_security.dm:35]: I purposefully left it as RPGMaker SOUL

**Comment by LemonInTheDark** [comment]
How does this look with runechat?

File: code/modules/jobs/job_types/head_of_security.dm (hunk 2 above)
Comment (AMonkeyThatCodes) [reply code/modules/jobs/job_types/head_of_security.dm:35]: ?

**Comment by AMonkeyThatCodes** [comment]
duping every single trim icon is dumb.

**Comment by AMonkeyThatCodes** [comment]
also this should be below players not above. Either that or it should offset runechat to spawn above it.

**Comment by MrMelbert** [comment]
Runechat is displayed above maptext

**Comment by LemonInTheDark** [comment]
Yeah but what does it look like, does it get cluttered?

**Comment by tralezab** [comment]
will show you, MAY move it or may not

File: code/modules/jobs/job_types/head_of_security.dm (hunk 2 above)
Comment (tralezab) [reply code/modules/jobs/job_types/head_of_security.dm:35]: its hard to quantify, just know town guard npcs are like a dime a dozen

<details>
  <summary>Spoiler warning</summary>
  
  ## SOUL
  
  
  
</details>

**Comment by Fikou** [comment]
bro i ran that event first 😭

File: code/modules/events/rpgtitles.dm [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+	card.update_icon()
+
+/**
+ * ### handle_current_jobs
+ *
+ * Calls on_crewmember_join on every crewmember
+ * If the item it is giving fantasy to is a storage item, there's a chance it'll drop in an item fortification scroll. neat!
+ */
+/datum/rpgtitle_controller/proc/handle_current_jobs()
+	for(var/mob/living/player in GLOB.player_list)
```
Comment (Fikou) [inline code/modules/events/rpgtitles.dm:84]: do as anything in the global living player list
also how does this work with AI's?

File: code/modules/events/rpgtitles.dm [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	else
+		for(var/biotype_flag in biotype_titles)
+			if(new_crewmember.mob_biotypes & biotype_flag)
+				maptext_title += "biotype_titles[biotype_flag] "
+		maptext_title = trim_right(maptext_title)
+		if(!maptext_title)
+			maptext_title = "Anomaly"
+
+	//mother of all strings...
+	new_crewmember.maptext = "<span class='maptext' style='text-align: center'><span style='color: [new_crewmember.chat_color || rgb(rand(100,255), rand(100,255), rand(100,255))]'>Level [rand(1, 100)] [maptext_title]</span></span>"
```
Comment (Fikou) [inline code/modules/events/rpgtitles.dm:59]: we have procs for generating random colors

File: code/modules/events/rpgtitles.dm (hunk 4 above)
Comment (tralezab) [reply code/modules/events/rpgtitles.dm:84]: adding to the player list is a /mob/ level proc sadly

ai get maptext but don't get ID changes

File: code/modules/jobs/job_types/head_of_security.dm (hunk 2 above)
Comment (AMonkeyThatCodes) [reply code/modules/jobs/job_types/head_of_security.dm:35]: what does that have to do with the bad name?

**Comment by Farquaar** [comment]
Is there a way to level up/level down?

**Comment by tralezab** [comment]
Not at the moment, there could be in theory tho

**Comment by NanoCats** [comment]
Do these do anything? Can you "level up"?

**Comment by LemonInTheDark** [comment]
https://github.com/tgstation/tgstation/pull/61579#issuecomment-923200570

File: code/modules/events/rpgtitles.dm (hunk 5 above)
Comment (tralezab) [reply code/modules/events/rpgtitles.dm:90]: it's not specific enough for what i'm looking for

Comment (Fikou) [reply code/modules/events/rpgtitles.dm:90]: alright

**Comment by ilikesaying** [comment]
That's how Mafia works

**Review by Watermelon914 — CHANGES_REQUESTED** [review]
You have a lot of duplicated sprites, could you possibly change them to be something different?

**Comment by tralezab** [comment]
added the special icons 👍

**Review by Watermelon914 — APPROVED** [review]
code looks fine, needs spritetainer review

**Review by Krysonism — APPROVED** [review]
The little icons are very charming.

I can't quite make out that the bartender one is, but the other ones are great.

**Pull request merged**